src/components/custom-field-input-date/custom-field-input-date.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import calendarSvg from '../../svgs/icon-calendar-fill.svg';
+import Icon from '../icon/icon.jsx';
+import CustomFieldInputText from '../custom-field-input-text/custom-field-input-text.jsx';
+import { convertToFormat, validDate } from './format/format-date.js';
+
+const isValidInput = (value) => {
+  if (value === '' || value === undefined) {
+    return true;
+  }
+
+  return validDate(value);
+};
+
+const isValueValid = (value, error, isInputValid = false) => {
+  if (error) {
+    return false;
+  }
+
+  if (!isInputValid) {
+    return false;
+  }
+
+  return isValidInput(value);
+};
+
+export default function CustomFieldInputDate(props) {
+  const inputRef = useRef(null);
+  const [isEditing, setIsEditing] = useState(true);
+  const [isValid, setIsValid] = useState(isValueValid(props.value, props.error, true));
+  const [isFocused, setIsFocused] = useState(false);
+
+  useEffect(() => {
+    const isInputValid = inputRef.current.validity.valid;
+    const valid = isValueValid(props.value, props.error, isInputValid);
+    setIsValid(valid);
+
+    if (!valid) {
+      setIsEditing(true);
+    } else {
+      setIsEditing(false);
+    }
+  }, [inputRef.current]);
+
+  useEffect(() => {
+    if (isEditing && isFocused) {
+      inputRef.current.focus();
+    }
+  }, [isEditing, isFocused]);
+
+  const handleOnFocus = () => {
+    setIsFocused(true);
+    setIsEditing(true);
+  };
+
+  const handleOnBlur = () => {
+    setIsFocused(false);
+    setIsEditing(false);
+  };
+
+  const handleOnChange = (event) => {
+    if (inputRef && inputRef.current) {
+      const isInputValid = inputRef.current.validity.valid;
+      const newDate = convertToFormat(event.target.value, 'yyyy-mm-dd');
+      setIsValid(isValueValid(newDate, props.error, isInputValid) && validDate(newDate));
+    }
+
+    props.onChange(event);
+  };
+
+  const helpText = () => {
+    if (!isValid && !isValidInput(props.value)) {
+      return `"${props.value}" is an invalid date`;
+    }
+
+    return props.helpText;
+  };
+
+  const sharedProps = {
+    className: props.className,
+    disabled: true,
+    icon: <Icon name={calendarSvg.id} title={props.label} stroke="primary" />,
+    label: props.label,
+    inputRef,
+    required: props.required,
+  };
+
+  if (isEditing || !isValid) {
+    const value = validDate(props.value) ? convertToFormat(props.value, 'yyyy-mm-dd') : props.value;
+
+    return (<CustomFieldInputText
+      {...sharedProps}
+      error={!isValid}
+      helpText={helpText()}
+      id={props.id}
+      key={`${props.id}-editing`}
+      min={convertToFormat(props.min, 'yyyy-mm-dd')}
```

Comment:
What do we think about simplifying the logic by making the `min` property to have a format `'yyyy-mm-dd'`? Same thing with `max`.

Reply:
I'd be down, but how do we enforce a format in Prop types?

Reply:
One of the things about prop-types is that we can provide a custom prop-type as a function:

```
  // You can also specify a custom validator. It should return an Error
  // object if the validation fails. Don't `console.warn` or throw, as this
  // won't work inside `oneOfType`.
  customProp: function(props, propName, componentName) {
    if (!/matchme/.test(props[propName])) {
      return new Error(
        'Invalid prop `' + propName + '` supplied to' +
        ' `' + componentName + '`. Validation failed.'
      );
    }
  },
```

See the documentation for "more" information: https://reactjs.org/docs/typechecking-with-proptypes.html

Reply:
I'll include this in the next PR since this is a readonly component and `min` and `max` are not handled here.

Reply:
See #84 for the implementation of this suggestion